src/documenting/snippets.md

```diff
@@ -0,0 +1,279 @@
+# Code snippets
+
+Iroha development happens around three important branches:
+[`dev`](https://github.com/hyperledger/iroha/tree/iroha2-dev),
+[`stable`](https://github.com/hyperledger/iroha/tree/iroha2-stable), and
+[`LTS`](https://github.com/hyperledger/iroha/tree/iroha2-lts).
+
+With that in mind, documenting all the API versions manually has
+limitations: at some point, the content in one of the branches will be
+different compared to the documentation. Moreover, code in the
+documentation may contain typos sometimes, and being able to run it won't
+be guaranteed without testing. This raises questions about showing the
+differences between branches for the new developers semi-automatically.
+
+Our solution is to use a custom syntax in the code comments to mark the
+snippets with the [dst-parser](https://github.com/soramitsu/dst-parser). We
+configure URLs in a custom JSON config file and automatically query the
+related URLs both on local builds and CI.
+
+In addition, we're also using a custom
+[markdown-it](https://github.com/markdown-it/markdown-it) plugin, so that
+we can easily include the resulting snippets.
+
+## Workflow
+
+### Preparing the requirements
+
+After you run `npm install` or `pnpm install`, a prebuilder script may run.
+It would ensure that an output directory for the snippets (`src/snippets`)
```

Comment:
So there is a possibility it doesn't ensure the existence of this directory?

Reply:
There was a discussion on potentially removing it.